CRM/Contact/BAO/Group.php

```diff
@@ -391,17 +388,8 @@ public static function create(&$params) {
       }
     }
     $group = new CRM_Contact_BAO_Group();
-    $group->copyValues($params);
-    //@todo very hacky fix for the fact this function wants to receive 'parents' as an array further down but
-    // needs it as a separated string for the DB. Preferred approaches are having the copyParams or save fn
-    // use metadata to translate the array to the appropriate DB type or altering the param in the api layer,
-    // or at least altering the param in same section as 'group_type' rather than repeating here. However, further down
-    // we need the $params one to be in it's original form & we are not sure what test coverage we have on that
-    if (isset($group->parents) && is_array($group->parents)) {
-      $group->parents = CRM_Core_DAO::VALUE_SEPARATOR . implode(CRM_Core_DAO::VALUE_SEPARATOR,
-          array_keys($group->parents)
-        ) . CRM_Core_DAO::VALUE_SEPARATOR;
-    }
```

Comment:
I'm confused about what type of serialization this field uses. In this old code it looks like it uses VALUE_SEPARATOR but in the metadata you specified COMMA.

Reply:
I did suggest it should be COMA :-) - check the comments on the PR summary - dang they disappeared / didn't save - I hate you githut.

Basically the BAO saves it as BOOKEND_SEPARATOR & then       CRM_Contact_BAO_GroupNestingCache::update(); overwrites those values with comma separated ones :-)

Reply:
Ok that is truly bizarre.